doc/man7/EVP_KDF_HKDF.pod

```diff
@@ -106,18 +82,39 @@ an error will occur.
 
 =back
 
-EVP_KDF_ctrl_str() type string: "mode"
-
-The value string is expected to be one of: "EXTRACT_AND_EXPAND", "EXTRACT_ONLY"
+The parameter is expected to be one of: "EXTRACT_AND_EXPAND", "EXTRACT_ONLY"
 or "EXPAND_ONLY".
 
+=item B<OSSL_KDF_PARAM_MODE_INT>
```

Comment:
Instead of a different name, it could be possible for `OSSL_KDF_PARAM_MODE` to be accepted both as a string and as an integer...  I see no compelling reason why that shouldn't be a possibility

Reply:
Which should be advertised by the settable call?

Reply:
I don't understand...

``` C
    {OSSL_KDF_PARAM_MODE, OSSL_PARAM_UNSIGNED_INTEGER, NULL, 0, 0},
    {OSSL_KDF_PARAM_MODE, OSSL_PARAM_OCTET_STRING, NULL, 0, 0},
```

(The `_from_text` functions may need to become smarter)

Reply:
`OSSL_PARAM_locate_const`?

It can be used on the returns of the query functions.

Reply:
Not sure what you're arguing at this point...  I gave you an example on how to get that advertised through the settable call, that was my main point.  Yes, there are consequences, and that's for another PR.

Reply:
I guess it comes down to which goes first?  That's the one that will be found most easily.

Having the back end code handle both is easy -- I originally wrote it that way in fact.

I'm happy to switch this to have a polymorphic type.  Well tomorrow.

Reply:
I'd like to see that switch happen.

Reply:
Sure.  I think I'll put the string version first since we're moving that way.

Reply:
Done